`ionic capacitor sync` command fails when providing platform

**Description:**
Hi, I have an Angular multiproject setup and use Ionic with Capacitor. I recently started migrating to Capacitor 4 and I have issues running a certain command from Ionic CLI since that version upgrade.

That command is `npx ionic capacitor sync android --no-build --project=app1`. I have the same issue when replacing `android` with `ios`.

If I remove `android` or `ios` from the command, it runs well but it syncs both platforms.

**Steps to Reproduce:**
Here is an example repo that mimics the structure of our multi-app project: [https://github.com/jbouchard24/ionic6-capacitor3-cli-angular-multiproject-bug-demo](https://github.com/jbouchard24/ionic6-capacitor3-cli-angular-multiproject-bug-demo)

Clone the repo, stay on the `master` branch (that uses Capacitor CLI 3.3.1), run `npm install` and then run `npx ionic capacitor sync android --no-build --project=app1` from the root of the project. You should see the following error:

<img width="595" alt="image" src="https://user-images.githubusercontent.com/54321952/186430315-e38ef0ea-ffcb-46b2-abec-789b26e640e0.png">

We found out that running Capacitor CLI <= 3.3.0 doesn't produce an error. You can see that behavior from the `working/capacitor-cli-3.3.0` branch of the demo repo.

**Output:**
```
  ionic:lib Terminal info: { ci: false, shell: 'C:\\WINDOWS\\system32\\cmd.exe', tty: true, windows: true } +0ms
  ionic:lib CLI global options: { _: [ 'capacitor', 'sync', 'android' ], help: null, h: null, verbose: true, quiet: null, interactive: true, color: true, confirm: null, json: null, build: false, project: 'app2', '--': [] } +3ms
  ionic:lib:project Project id from args: app2 +0ms
  ionic:lib:project Project type from config: @ionic/angular (angular) +1ms
  ionic:lib:project Project details: { context: 'multiapp', id: 'app2', type: 'angular', errors: [], configPath: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\ionic.config.json' } +0ms
  ionic Context: { binPath: 'C:\\Users\\JocelynBouchard\\AppData\\Roaming\\nvm\\v14.16.1\\node_modules\\@ionic\\cli\\bin\\ionic', libPath: 'C:\\Users\\JocelynBouchard\\AppData\\Roaming\\nvm\\v14.16.1\\node_modules\\@ionic\\cli', execPath: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo', version: '6.20.1' } +0ms
  ionic:lib:integrations:capacitor Getting config with Capacitor CLI: [ 'config', '--json' ] +0ms
> npm.cmd i -E @capacitor/android@3.3.1
npm WARN @capacitor/ios@3.3.1 requires a peer of @capacitor/core@^3.3.0 but none is installed. You must install peer dependencies yourself.
npm WARN @capacitor/android@3.3.1 requires a peer of @capacitor/core@^3.3.0 but none is installed. You must install peer dependencies yourself.
npm WARN app2@1.0.0 No description
npm WARN app2@1.0.0 No repository field.

+ @capacitor/android@3.3.1
updated 1 package and audited 2 packages in 1.054s
found 0 vulnerabilities

> capacitor.cmd add android
[capacitor] 2022-08-24T13:30:05.599Z capacitor:config config: {
[capacitor]   android: {
[capacitor]     name: 'android',
[capacitor]     minVersion: '21',
[capacitor]     studioPath: LazyPromise [Promise] {
[capacitor]       <pending>,
[capacitor]       _executor: [AsyncFunction (anonymous)]
[capacitor]     },
[capacitor]     platformDir: 'android',
[capacitor]     platformDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android',
[capacitor]     cordovaPluginsDir: 'capacitor-cordova-android-plugins',
[capacitor]     cordovaPluginsDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\capacitor-cordova-android-plugins',
[capacitor]     appDir: 'app',
[capacitor]     appDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app',
[capacitor]     srcDir: 'app/src',
[capacitor]     srcDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app\\src',
[capacitor]     srcMainDir: 'app/src/main',
[capacitor]     srcMainDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app\\src\\main',
[capacitor]     assetsDir: 'app/src/main/assets',
[capacitor]     assetsDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app\\src\\main\\assets',
[capacitor]     webDir: 'app/src/main/assets/public',
[capacitor]     webDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app\\src\\main\\assets\\public',
[capacitor]     resDir: 'app/src/main/res',
[capacitor]     resDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app\\src\\main\\res',
[capacitor]     apkName: 'app-debug.apk',
[capacitor]     buildOutputDir: 'app/build/outputs/apk//debug',
[capacitor]     buildOutputDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\android\\app\\build\\outputs\\apk\\debug'
[capacitor]   },
[capacitor]   ios: {
[capacitor]     name: 'ios',
[capacitor]     minVersion: '12.0',
[capacitor]     platformDir: 'ios',
[capacitor]     platformDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\ios',
[capacitor]     scheme: 'App',
[capacitor]     cordovaPluginsDir: 'capacitor-cordova-ios-plugins',
[capacitor]     cordovaPluginsDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\ios\\capacitor-cordova-ios-plugins',
[capacitor]     nativeProjectDir: 'App',
[capacitor]     nativeProjectDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\ios\\App',
[capacitor]     nativeTargetDir: 'App/App',
[capacitor]     nativeTargetDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\ios\\App\\App',
[capacitor]     nativeXcodeProjDir: 'App/App.xcodeproj',
[capacitor]     nativeXcodeProjDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\ios\\App\\App.xcodeproj',
[capacitor]     nativeXcodeWorkspaceDir: LazyPromise [Promise] {
[capacitor]       <pending>,
[capacitor]       _executor: [AsyncFunction (anonymous)]
[capacitor]     },
[capacitor]     nativeXcodeWorkspaceDirAbs: LazyPromise [Promise] {
[capacitor]       <pending>,
[capacitor]       _executor: [AsyncFunction (anonymous)]
[capacitor]     },
[capacitor]     webDir: LazyPromise [Promise] {
[capacitor]       <pending>,
[capacitor]       _executor: [AsyncFunction (anonymous)]
[capacitor]     },
[capacitor]     webDirAbs: LazyPromise [Promise] {
[capacitor]       <pending>,
[capacitor]       _executor: [AsyncFunction (anonymous)]
[capacitor]     },
[capacitor]     podPath: 'pod'
[capacitor]   },
[capacitor]   web: {
[capacitor]     name: 'web',
[capacitor]     platformDir: '../../www/app2',
[capacitor]     platformDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\www\\app2'
[capacitor]   },
[capacitor]   cli: {
[capacitor]     rootDir: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\node_modules\\@capacitor\\cli',
[capacitor]     assetsDir: 'assets',
[capacitor]     assetsDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\node_modules\\@capacitor\\cli\\assets',
[capacitor]     assets: { ios: [Object], android: [Object] },
[capacitor]     package: {
[capacitor]       _args: [Array],
[capacitor]       _development: true,
[capacitor]       _from: '@capacitor/cli@3.3.1',
[capacitor]       _id: '@capacitor/cli@3.3.1',
[capacitor]       _inBundle: false,
[capacitor]       _integrity: 'sha512-PWGWhK/++NPziLigzO0RVORif+V7DAhm99Sd+2UuhfK40elQYK80/DeqCGwh5dVYwuiPN6Q8U2q9mz+mx8LM0g==',
[capacitor]       _location: '/@capacitor/cli',
[capacitor]       _phantomChildren: [Object],
[capacitor]       _requested: [Object],
[capacitor]       _requiredBy: [Array],
[capacitor]       _resolved: 'https://registry.npmjs.org/@capacitor/cli/-/cli-3.3.1.tgz',
[capacitor]       _spec: '3.3.1',
[capacitor]       _where: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo',
[capacitor]       author: [Object],
[capacitor]       bin: [Object],
[capacitor]       bugs: [Object],
[capacitor]       dependencies: [Object],
[capacitor]       description: 'Capacitor: Cross-platform apps with JavaScript and the web',
[capacitor]       devDependencies: [Object],
[capacitor]       engines: [Object],
[capacitor]       files: [Array],
[capacitor]       gitHead: '8ff7434e9b775e6acb05efae23771ee1b18e26c9',
[capacitor]       homepage: 'https://capacitorjs.com',
[capacitor]       jest: [Object],
[capacitor]       keywords: [Array],
[capacitor]       license: 'MIT',
[capacitor]       main: 'dist/index.js',
[capacitor]       name: '@capacitor/cli',
[capacitor]       publishConfig: [Object],
[capacitor]       repository: [Object],
[capacitor]       scripts: [Object],
[capacitor]       types: 'dist/declarations.d.ts',
[capacitor]       version: '3.3.1'
[capacitor]     },
[capacitor]     os: 'windows'
[capacitor]   },
[capacitor]   app: {
[capacitor]     rootDir: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2',
[capacitor]     appId: 'io.ionic.app2',
[capacitor]     appName: 'app2',
[capacitor]     webDir: '../../www/app2',
[capacitor]     webDirAbs: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\www\\app2',
[capacitor]     package: {
[capacitor]       name: 'app2',
[capacitor]       version: '1.0.0',
[capacitor]       description: '',
[capacitor]       author: '',
[capacitor]       license: 'ISC',
[capacitor]       dependencies: [Object],
[capacitor]       devDependencies: [Object]
[capacitor]     },
[capacitor]     extConfigType: 'json',
[capacitor]     extConfigName: 'capacitor.config.json',
[capacitor]     extConfigFilePath: 'C:\\work\\_other\\ionic6-capacitor3-cli-angular-multiproject-bug-demo\\projects\\app2\\capacitor.config.json',
[capacitor]     extConfig: {
[capacitor]       appId: 'io.ionic.app2',
[capacitor]       appName: 'app2',
[capacitor]       webDir: '../../www/app2',
[capacitor]       bundledWebRuntime: false
[capacitor]     },
[capacitor]     bundledWebRuntime: false
[capacitor]   }
[capacitor] }
[capacitor] [error] android platform already exists.
[capacitor]         To re-add this platform, first remove .\android, then run this command again.
[capacitor]         WARNING: Your native project will be completely removed.
[ERROR] An error occurred while running subprocess capacitor.

        capacitor.cmd add android exited with exit code 1.

        Re-running this command with the --verbose flag may provide more information.
  ionic:utils-process onBeforeExit handler: 'process.exit' received +0ms
  ionic:utils-process onBeforeExit handler: running 2 functions +0ms
  ionic:utils-process error while killing process tree for 3728: Error: Command failed: taskkill /pid 3728 /T /F
  ionic:utils-process ERROR: The process "3728" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (child_process.js:308:12)
  ionic:utils-process     at ChildProcess.emit (events.js:315:20)
  ionic:utils-process     at maybeClose (internal/child_process.js:1048:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5) {
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 3728 /T /F'
  ionic:utils-process } +196ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 3728 /T /F
  ionic:utils-process ERROR: The process "3728" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (child_process.js:308:12)
  ionic:utils-process     at ChildProcess.emit (events.js:315:20)
  ionic:utils-process     at maybeClose (internal/child_process.js:1048:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5) {
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 3728 /T /F'
  ionic:utils-process } +2ms
  ionic:utils-process error while killing process tree for 35304: Error: Command failed: taskkill /pid 35304 /T /F
  ionic:utils-process ERROR: The process "35304" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (child_process.js:308:12)
  ionic:utils-process     at ChildProcess.emit (events.js:315:20)
  ionic:utils-process     at maybeClose (internal/child_process.js:1048:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5) {
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 35304 /T /F'
  ionic:utils-process } +2ms
  ionic:utils-process onBeforeExit handler: error from function: Error: Command failed: taskkill /pid 35304 /T /F
  ionic:utils-process ERROR: The process "35304" not found.
  ionic:utils-process 
  ionic:utils-process     at ChildProcess.exithandler (child_process.js:308:12)
  ionic:utils-process     at ChildProcess.emit (events.js:315:20)
  ionic:utils-process     at maybeClose (internal/child_process.js:1048:16)
  ionic:utils-process     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5) {
  ionic:utils-process   killed: false,
  ionic:utils-process   code: 128,
  ionic:utils-process   signal: null,
  ionic:utils-process   cmd: 'taskkill /pid 35304 /T /F'
  ionic:utils-process } +1ms
  ionic:utils-process processExit: exiting (exit code: 1) +2ms
```

**My `ionic info`:**

```
Ionic:

   Ionic CLI                     : 6.20.1
   Ionic Framework               : @ionic/angular 6.2.3
   @angular-devkit/build-angular : 14.1.3
   @angular-devkit/schematics    :  14.1.3
   @angular/cli                  : 14.1.3
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.3.1
   @capacitor/android : 3.3.1
   @capacitor/core    : 3.3.1
   @capacitor/ios     : 3.3.1

Utility:

   cordova-res : 0.15.4
   native-run  : not installed globally

System:

   NodeJS : v14.16.1
   npm    : 6.14.12
   OS     : Windows 11
```

Thanks!

--- a/packages/@ionic/cli/src/lib/integrations/capacitor/index.ts
+++ b/packages/@ionic/cli/src/lib/integrations/capacitor/index.ts
@@ -195,7 +195,7 @@ export class Integration extends BaseIntegration<ProjectIntegration> {
       return new Promise((resolve, reject) => {
         exec(cmd, opts, (error, stdout, stderr) => {
           if (error) {
-            resolve(undefined);
+            reject();
           }
           resolve(stdout);
         })

--- a/packages/@ionic/cli/src/lib/integrations/capacitor/index.ts
+++ b/packages/@ionic/cli/src/lib/integrations/capacitor/index.ts
@@ -191,8 +191,6 @@ export class Integration extends BaseIntegration<ProjectIntegration> {
 
     debug('Getting config with Capacitor CLI: %O', args);
 
-    //const output = await this.e.shell.cmdinfo('capacitor', args, { cwd: this.root });
-
     const output = await ((cmd: string, opts: ExecOptions): Promise<string | undefined> => {
       return new Promise((resolve, reject) => {
         exec(cmd, opts, (error, stdout, stderr) => {

--- a/packages/@ionic/cli/src/lib/integrations/capacitor/index.ts
+++ b/packages/@ionic/cli/src/lib/integrations/capacitor/index.ts
@@ -1,3 +1,4 @@
+import { exec, ExecOptions } from 'child_process';
 import { parseArgs } from '@ionic/cli-framework';
 import { mkdirp, pathExists } from '@ionic/utils-fs';
 import { prettyPath } from '@ionic/utils-terminal';
@@ -190,7 +191,18 @@ export class Integration extends BaseIntegration<ProjectIntegration> {
 
     debug('Getting config with Capacitor CLI: %O', args);
 
-    const output = await this.e.shell.cmdinfo('capacitor', args, { cwd: this.root });
+    //const output = await this.e.shell.cmdinfo('capacitor', args, { cwd: this.root });
+
+    const output = await ((cmd: string, opts: ExecOptions): Promise<string | undefined> => {
+      return new Promise((resolve, reject) => {
+        exec(cmd, opts, (error, stdout, stderr) => {
+          if (error) {
+            resolve(undefined);
+          }
+          resolve(stdout);
+        })
+      })
+    })(`capacitor ${args.join(' ')}`, { cwd: this.root });
 
     if (!output) {
       debug('Could not get config from Capacitor CLI (probably old version)');

--- a/packages/@ionic/lab/package.json
+++ b/packages/@ionic/lab/package.json
@@ -44,13 +44,14 @@
     "@ionic/cli-framework": "5.1.3",
     "@ionic/utils-fs": "3.1.6",
     "chalk": "^4.0.0",
-    "express": "^4.16.2",
+    "express": "4.16.2",
     "tslib": "^2.0.1"
   },
   "devDependencies": {
     "@ionic-internal/ionic-ds": "^2.1.0",
     "@stencil/core": "~1.8.5",
-    "@types/express": "^4.11.0",
+    "@types/express": "4.11.0",
+    "@types/express-serve-static-core": "4.11.0",
     "@types/node": "~10.17.13",
     "lint-staged": "^10.0.2",
     "rimraf": "^3.0.0",

--- a/packages/@ionic/v1-toolkit/package.json
+++ b/packages/@ionic/v1-toolkit/package.json
@@ -42,7 +42,7 @@
     "chalk": "^4.0.0",
     "chokidar": "^3.0.1",
     "debug": "^4.0.0",
-    "express": "^4.16.2",
+    "express": "4.16.2",
     "gulp": "^4.0.2",
     "http-proxy-middleware": "^0.20.0",
     "tiny-lr": "^1.1.0",
@@ -51,7 +51,8 @@
   },
   "devDependencies": {
     "@types/debug": "^4.1.1",
-    "@types/express": "^4.11.0",
+    "@types/express": "4.11.0",
+    "@types/express-serve-static-core": "4.11.0",
     "@types/gulp": "^4.0.6",
     "@types/http-proxy-middleware": "^0.19.0",
     "@types/jest": "^26.0.10",

